### PR TITLE
ci: push release image when there is a tag

### PR DESCRIPTION
**Pull request by zhujian7**
The post submit workflow will build and push the addon manager image with "latest" tag, only the main branch's code is the latest, stop other branches.

### CODE REVIEW

**Comment by qiujian16** [comment]
I think we need this for release branch

**Comment by zhujian7** [comment]
> I think we need this for release branch

yes, you are right. for release branches, we only need to build and push the image when there is a new tag, right? I will change that.

**Comment by qiujian16** [comment]
yes, I think so

**Review by qiujian16 — COMMENTED** [review]
/approve
/lgtm

File: .github/workflows/go-release.yml [hunk 1]
```diff
@@ -31,13 +31,83 @@ jobs:
...
+          echo "# Addon Framework ${{ needs.env.outputs.RELEASE_VERSION }}" > /home/runner/work/changelog.txt
+          echo "- See the [CHANGELOG](https://github.com/open-cluster-management-io/addon-framework/blob/main/CHANGELOG/CHANGELOG-${{ needs.env.outputs.MAJOR_RELEASE_VERSION }}.md) for more details." >> /home/runner/work/changelog.txt
+          echo "- The released image is quay.io/open-cluster-management/addon-manager:${{ needs.env.outputs.RELEASE_VERSION }}" >> /home/runner/work/changelog.txt
       - name: publish release
         uses: softprops/action-gh-release@v0.1.5
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         with:
-          body_path: /home/runner/addon-framework/changelog.txt
+          body_path: /home/runner/work/changelog.txt
```
Comment (zhujian7) [inline .github/workflows/go-release.yml:113]: The content is written to `/home/runner/work/changelog.txt`

**Comment by zhujian7** [comment]
rebase https://github.com/open-cluster-management-io/addon-framework/pull/162 to fix the CI problem.

**Pull request merged**